azurelinuxagent/pa/provision/default.py

```diff
@@ -83,9 +83,10 @@ def run(self):
 
             self.provision(ovf_env)
 
-            thumbprint = self.reg_ssh_host_key()
-            self.osutil.restart_ssh_service()
-
+            if conf.get_regenerate_ssh_host_key():
+                self.reg_ssh_host_key()
+                self.osutil.restart_ssh_service()
+            thumbprint = self.get_ssh_host_key_thumbprint()
```

Comment:
This change probably makes the mainline path slower, since "thumbprint" can't be retrieved from the ssh service until that service is fully up and running. It would be smarter to hold on to the thumbprint and use it in the mainline code path. That is...

```python
thumbprint = ""

if conf.get_regenerate_ssh_host_key():
    thumbprint = self.reg_ssh_host_key()
    self.osutil.restart_ssh_service()
else
    thumbprint = self.get_ssh_host_key_thumbprint()
```